src/lib/fb.js

```diff
@@ -1,36 +1,33 @@
 import {firebaseConfig} from 'webdev_config';
 import {store} from './store';
 import {clearSignedInState} from './actions';
-import firestoreLoader from './firestore-loader';
+import firebaseLoader from './utils/firebase-loader';
 
-/* eslint-disable require-jsdoc */
-
-firebase.initializeApp(firebaseConfig);
-
-// Initialize performance monitoring
-firebase.performance();
+const firebasePromise = firebaseLoader('app', 'auth', 'performance');
+firebasePromise.then(initialize).catch((err) => {
+  console.error('failed to load Firebase', err);
+});
 
-// Listen for the user's signed in state and update the store.
-let firestoreUserUnsubscribe = null;
-firebase.auth().onAuthStateChanged((user) => {
-  store.setState({checkingSignedInState: false});
+const firestoreLoader = (() => {
```

Comment:
> So we've always dynamically loaded Firestore, because we don't need it unless the user is signed in.

That makes sense.

> And the existing model does await whenever we need it (to write/clear the user's measure row), but that will always be a no-op after we sign in.

I do wonder if that means we're going to have a lot of boilerplate `await firebase` calls as we start to use fb more. Not a huge issue for this PR but worth considering.